### PR TITLE
Escape Windows backslash when outputing to markdown

### LINKED ISSUE #2272: Missing backslash in printed CARGO_HOME path on Windows
**Problem**

I installed Rust using rustup under Windows 10 and noticed a missing backslash in the post install message which tells you the Cargo bin directory:

```
Rust is installed now. Great!

To get started you need Cargo's bin directory (%USERPROFILE%.cargo\bin) in your
PATH
environment variable. Future applications will automatically have the
correct environment, but you may need to restart your current shell.

Press the Enter key to continue.
```

It should be `%USERPROFILE%\.cargo\bin` with a backslash after `%USERPROFILE%` to be valid.

Note that it printed the following earlier:

```
It will add the cargo, rustc, rustup and other commands to
Cargo's bin directory, located at:

  C:\Users\Simran\.cargo\bin
```

And it added this to the PATH environment variable correctly.

**Steps**

1. [Download rustup-init.exe (64-bit)](https://static.rust-lang.org/rustup/dist/x86_64-pc-windows-msvc/rustup-init.exe)
2. `1) Proceed with installation (default)`
3. Wait until installation completes and check the last block of messages

**Possible Solution(s)**

The message is coming from here:

_src\cli\self_update.rs_

```rust
macro_rules! post_install_msg_win {
    () => {
        r"# Rust is installed now. Great!

To get started you need Cargo's bin directory ({cargo_home}\\bin) in your `PATH`
environment variable. Future applications will automatically have the
correct environment, but you may need to restart your current shell.
"
    };
}
```

The backslash before `bin` (which shows correctly) is escaped, but the rest is coming from the variable `cargo_home`. Interestingly, a different variable is used for the other message, `cargo_home_bin`:

```rust
macro_rules! pre_install_msg_template {
…
It will add the `cargo`, `rustc`, `rustup` and other commands to
Cargo's bin directory, located at:

    {cargo_home_bin}
```

It could be used instead of `{cargo_home}\\bin` I guess, but with the env var `%USERPROFILE%` resolved. I'm not sure how else to fix the problem. The following looks alright using a raw string, removing the need for escaping the backslash:

```rust
fn canonical_cargo_home() -> Result<String> {
…
            path_str = String::from(r"%USERPROFILE%\.cargo");
```

Or does it need to be double-escaped to `r"%USERPROFILE%\\.cargo"` (non-raw `"%USERPROFILE%\\\\.cargo"` I suppose)?

**Notes**

Output of `rustup --version`: rustup 1.21.1 (7832b2ebe 2019-12-20)

Output of `rustup show`:

```
Default host: x86_64-pc-windows-msvc
rustup home:  C:\Users\Simran\.rustup

stable-x86_64-pc-windows-msvc (default)
rustc 1.42.0 (b8cedc004 2020-03-09)
```

### PR DESCRIPTION
**Pull request by tesuji**
Closes #2272
Duplicate close #2174

### CODE REVIEW

**Comment by tesuji** [comment]
Result from windows: https://github.com/rust-lang/rustup/pull/2274/checks?check_run_id=543340236#step:15:21

**Review by rbtcollins — APPROVED** [review]
LGTM though I'm not sure if the variation between windows and unix needs to be as large as it is.

File: ci/actions-templates/windows-builds-template.yaml
Comment (kinnison) [inline ci/actions-templates/windows-builds-template.yaml]: It's not clear to me what your goal is here.  Why are you installing the built rustup into the test host at this time?
Given this is part of a `do not merge` commit, did you mean to leave it in?

File: src/cli/self_update.rs
Comment (kinnison) [inline src/cli/self_update.rs:69]: This commit seems unrelated to your other work, why did you do it?

**Review by kinnison — CHANGES_REQUESTED** [review]
There are `[do not merge]`  commits in this sequence.  Please rebase them away.

File: ci/actions-templates/windows-builds-template.yaml
Comment (tesuji) [reply ci/actions-templates/windows-builds-template.yaml]: I don't have a Windows machine and I don't wanna run the virtual machine.
So to check if the new installation prompt for Windows displays correctly, 
I add this commit.

File: src/cli/self_update.rs
Comment (tesuji) [reply src/cli/self_update.rs:69]: The macro argument of that macro is always literal to me.
I'd move it to another PR (this change is so small) if you prefer.

**Pull request merged**
